src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/process-form.service.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { TestBed } from "@angular/core/testing";
+import { ProcessFormService } from "./process-form.service";
+import { FormArray, FormControl, FormGroup } from "@angular/forms";
+
+describe("ProcessFormService", () => {
```

Comment:
I see that we cover the lines, but we don;'t test the logic much.
Where are the tests for the real form transformation?
Some form state before, apply the transformation, assert the output is as expected.

Reply:
Added logic transformation unit tests.